### ISSUE TITLE
🐛 Bug Report: [Accessibility ] Catalog Table Sort icon screen reader issue

### ISSUE THREAD

**Issue by DorisCCLin** [backstage/backstage#18693]
### 📜 Description

The sort icon on the table can only be visible when mouse-over. However even tho the element can be focused, the screen reader can't read it properly without proper tagging. 

### 👍 Expected behavior

Screen reader should be able to read it, also reconsider mouse-over might not be the best option for Accessibility users.

### 👎 Actual Behavior with Screenshots

![Screen Shot 2023-07-17 at 11 23 51 AM](https://github.com/backstage/backstage/assets/27718001/ed4ffa5a-3e05-46af-b896-915795e01e45)

### 👟 Reproduction steps

1. go to https://demo.backstage.io/catalog?filters%5Bkind%5D=component&filters%5Buser%5D=owned
2. hover the mouse on any top row of each column to see the sort icon
3. Use a screen reader, the sort feature is not stated. 

### 🖥️ Your Environment

OS:   Darwin 21.6.0 - darwin/arm64
node: v18.15.0
yarn: 1.22.19
cli:  0.22.4 (installed)
backstage:  1.11.1

### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

Yes I am willing to submit a PR!

**Comment by benjdlambert** [comment]
Is this something you want to contribute and help us out with @DorisCCLin? :pray:

**Comment by npiyush97** [comment]
@benjdlambert fixed in later version wanna upgrade ? 
https://material-table-core.github.io/docs/material-ui-v5

**Comment by npiyush97** [comment]
@tudi2d Sorry, if I was not clear this will be fixed if we upgrade material-table/core:^3.1.0 to 6.3.1 minimum to fix this issue or switch over to mui table.
